### ISSUE TITLE
Chat down't work - app-1 reports: ERROR web_server::errors: 422 unprocessable Entity , error decoding response body

### ISSUE THREAD

**Issue by thfrei** [bionic-gpt/bionic-gpt#491]
### Describe your issue

When I submit a chat message to an AI assistant using a dataset, it goes into endless loop. Server logs show this error message:

```
app-1           | 2024-05-23T07:26:31.509188Z ERROR web_server::errors: response="status = 422 Unprocessable Entity, message = error decoding response body"                                                                  
barricade-1     | [2024-05-23T07:26:31Z INFO  actix_web::middleware::logger] 10.6.198.115 "POST /completions/5 HTTP/1.1" 200 73 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.193270                                                                  
barricade-1     | [2024-05-23T07:26:32Z INFO  actix_web::middleware::logger] 10.6.198.115 "POST /app/team/1/update_response HTTP/1.1" 303 0 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.001817                                                      
barricade-1     | [2024-05-23T07:26:32Z INFO  actix_web::middleware::logger] 10.6.198.115 "GET /app/team/1/console/38 HTTP/1.1" 200 10150 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.002476
app-1           | 2024-05-23T07:26:40.881233Z ERROR web_server::errors: response="status = 422 Unprocessable Entity, message = error decoding response body"                                                                  
barricade-1     | [2024-05-23T07:26:40Z INFO  actix_web::middleware::logger] 10.6.198.115 "POST /completions/5 HTTP/1.1" 200 73 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.184773                                                                  
barricade-1     | [2024-05-23T07:26:41Z INFO  actix_web::middleware::logger] 10.6.198.115 "POST /app/team/1/update_response HTTP/1.1" 303 0 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.002077                                                      
barricade-1     | [2024-05-23T07:26:41Z INFO  actix_web::middleware::logger] 10.6.198.115 "GET /app/team/1/console/38 HTTP/1.1" 200 10150 "http://10.253.2.27:3001/app/team/1/console/38" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0" 0.003463                 
```

### Describe your setup/startup scripts

docker compose.yml

```
# cat docker-compose.yml
# Do not expose this configuration to the internet. This is an
# example configuration so that you can try out Bionic GPT.
#
# This is NOT a production configuration.
services:

  #llm-api:
  #  image: ghcr.io/bionic-gpt/llama-3-8b-chat:1.1.1
  #  platform: linux/amd64

  #embeddings-api:
  #  image: ghcr.io/huggingface/text-embeddings-inference:cpu-0.6
  #  platform: linux/amd64
  #  command: --model-id BAAI/bge-small-en-v1.5

  # Document parsing, OCR and chunking.
  # chunking-engine:
  unstructured: ##### <<<<<==== here it would not process any documents if it was still chunking-engine
    image: downloads.unstructured.io/unstructured-io/unstructured-api:4ffd8bc
    platform: linux/amd64

  # Postgres pre-loaded with pgVector
  # To connect outside docker `docker compose exec postgres psql -U postgres`
  postgres:
    image: ankane/pgvector
    platform: linux/amd64
    environment:
      POSTGRES_PASSWORD: password1
      POSTGRES_USER: postgres
      POSTGRES_DB: keycloak
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  # Configure Authn Proxy to listen on port 9090 and reverse proxy nginx on port 80.
  barricade:
    image: purtontech/barricade:1.2.9
    environment:
        # This secret key is used to encrypt cookies.
        SECRET_KEY: 190a5bf4b3x0f4f5d949
        DATABASE_URL: postgresql://postgres:password1@postgres:5432/bionic-gpt?sslmode=disable
        FORWARD_URL: app
        FORWARD_PORT: 7703
        PORT: 3000
        # Any requests that meet the following regular expressions
        # with pass through. i.e. They don't require auth.
        SKIP_AUTH_FOR: /static*
        REDIRECT_URL: '/'
        USER_TABLE_NAME: 'barricade_users'
        MAX_PAYLOAD_SIZE: 4000000
    ports:
      - "3001:3000"
    depends_on:
      postgres:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully

  # Sets up our database tables
  migrations:
    image: ghcr.io/bionic-gpt/bionicgpt-db-migrations:1.7.15
    platform: linux/amd64
    environment:
      DATABASE_URL: postgresql://postgres:password1@postgres:5432/bionic-gpt?sslmode=disable
    depends_on:
      postgres:
        condition: service_healthy

  # Parses documents into chunks and creates embeddings.
  pipeline-job:
    image: ghcr.io/bionic-gpt/bionicgpt-pipeline-job:1.7.15
    platform: linux/amd64
    environment:
      APP_DATABASE_URL: postgresql://postgres:password1@postgres:5432/bionic-gpt?sslmode=disable
    depends_on:
      postgres:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully

  # Our axum server delivering our user interface
  app:
    image: ghcr.io/bionic-gpt/bionicgpt:1.7.15
    platform: linux/amd64
    environment:
      SIGNOUT_URL: /auth/sign_out
      APP_DATABASE_URL: postgresql://postgres:password1@postgres:5432/bionic-gpt?sslmode=disable
    depends_on:
      postgres:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully
```

### Steps to reproduce

1. configure openai
2. Upload markdown Documents
3. ask chat question

### What was the expected result?

get an answer

### Any additional information

Openai is configured correctly i guess, since it can embedd each document and i get an answer when not using any dataset.

Thank you for your help.
PS: Thanks for bringing back the docker-compose.yml. I wondered where it went, last time I checked the repo half a year ago.

**Comment by 9876691** [comment]
Hi @thfrei Thanks for the detailed report.

I've put a fix in to change the endpoint it's looking for to unstructured and I'll test it.

We removed the docker compose as we were maintaining a lot of config which causes maintenance issue.

We brought it back, because for some people the Kubernetes install is a bit too much.

**Comment by 9876691** [comment]
Doc upload for `docker-compose` is fixed in 1.7.16
